packages/shared/src/lib/cookie.ts

```diff
@@ -10,11 +10,15 @@ export type CookieOptions = {
   secure: boolean;
 };
 
-export const getCookies = (
-  names: string[],
-): Record<string, string> | undefined => {
+export const getCookies = <
+  T extends string[] = string[],
+  R = Record<T[number], string>,
+>(
```

Comment:
Just improved the typings so that the returned object would contain the right keys from the type.